packages/hoprd/src/api/v2/paths/channel/list.ts

```diff
@@ -0,0 +1,87 @@
+import type { Operation } from 'express-openapi'
+import type Hopr from '@hoprnet/hopr-core'
+import { ChannelStatus, PublicKey } from '@hoprnet/hopr-utils'
+import { STATUS_CODES } from '../../'
+
+export interface ChannelInfo {
+  type: 'outgoing' | 'incoming'
+  channelId: string
+  peerId: string
+  status: string
+  balance: string
+}
+
+const channelStatusToString = (status: ChannelStatus): string => {
+  if (status === 0) return 'Closed'
+  else if (status === 1) return 'WaitingForCommitment'
+  else if (status === 2) return 'Open'
+  else if (status === 3) return 'PendingToClose'
+  return 'Unknown'
+}
+
+export const listChannels = async (node: Hopr) => {
+  const selfPubKey = new PublicKey(node.getId().pubKey.marshal())
+  const selfAddress = selfPubKey.toAddress()
+
+  const channelsFrom: ChannelInfo[] = (await node.getChannelsFrom(selfAddress))
+    .filter((channel) => channel.status !== ChannelStatus.Closed)
```

Comment:
This will filter out closed channels, but we may want to get all of them, can we add a parameter to allow querying all channels?